catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsSecuritySupport.java

```diff
@@ -37,4 +43,30 @@ void buildPolicyMap(HashMap<String, Set<String>> policyMap,
             }
         }
     }
+
+    /**
+     * Creates and returns a map of properties containing a security subject. If the correct subject
+     * type can not be obtained from the passed in subject or from the thread context no subject will
+     * be added to the properties.
+     * If the passed in subject is not compatible the method will fallback to trying to pull it from
+     * the thread context.
+     * @param subject The subject to be added. Can be {@code null}
+     */
+    Map<String, Serializable> getPropertiesWithSubject(Object subject) {
```

Comment:
I agree. I wrote this method a few different ways but ended with this to minimize duplicating code. One of the ways was to just return the subject for a given operation. I will go back to that way of doing it.